src/Engine/GameObject/Pathfinding.cpp

```diff
@@ -1,26 +1,96 @@
 #include "Pathfinding.hpp"
+#include "Scene\Scene.hpp"
 
-std::vector<char> PathfindingMap::Neighbors(char id) {
-	return edges[id];
+#define MAXPATHNODE 1024
+
+static PathNode* pathnodes[MAXPATHNODE];
+static int numPathNodes = 0;
+
+int PathNode::nextId = 0;
+
+bool PathNode::operator==(const PathNode& target) {
+	return this->x == target.x && this->y == target.y;
+}
+
+void PathNode::Create(double x, double y) {
+	this->createId();
+	this->x = x;
+	this->y = y;
 }
 
-void BreadthFirstSearch(PathfindingMap map, char start) {
-	std::queue<char> frontier;
-	frontier.push(start);
+void PathNode::createId() {
+	this->id = PathNode::nextId;
+	pathnodes[this->id] = this;
+	PathNode::nextId++;
+}
+
+PathNode* PathNode::getNode(UID_t id) {
+	return pathnodes[id];
+}
 
-	std::unordered_set<char> visited;
-	visited.insert(start);
+double PathNode::heuristic(UID_t pointA, UID_t pointB) {
+	return std::abs(pathnodes[pointA]->x - pathnodes[pointB]->x) + std::abs(pathnodes[pointA]->y - pathnodes[pointB]->y);
+}
+
+std::vector<UID_t> PathfindingMap::Neighbors(UID_t id) {
+	return edges[id];
+}
 
+std::unordered_map<UID_t, UID_t> pathSearch(PathfindingMap map, UID_t start, UID_t goal) {
+	PriorityQueue<UID_t, priority_t> frontier;
+	frontier.put(start, 0);
+	std::unordered_map<UID_t, UID_t> cameFrom;
+	cameFrom[start] = start;
```

Comment:
nit: could we name this prev / prevNode?

Reply:
No